tests/test_jira.py

```diff
@@ -0,0 +1,380 @@
+
+    def setUp(self):
+        self.tmp_path = tempfile.mkdtemp(prefix='perceval_')
+
+    def tearDown(self):
+        shutil.rmtree(self.tmp_path)
+
+    @httpretty.activate
+    def test_fetch_from_cache(self):
+        """ Test whether a list of issues is returned from cache """
+
+        bodies_json = read_file('data/jira/jira_issues_page_2.json')
+        expected_json = read_file('data/jira/jira_issues_fetch_expected_page_3.json')
+
+        httpretty.register_uri(httpretty.GET,
+                               JIRA_SEARCH_URL,
+                               body=bodies_json, status=200)
+
+        # First, we fetch the issues from the server, storing them
+        # in a cache
+        cache = Cache(self.tmp_path)
+        jira = Jira(JIRA_SERVER_URL, cache=cache)
+
+        issues = [issue for issue in jira.fetch()]
+        del issues[0]['timestamp']
+
+        # Now, we get the issues from the cache.
+        # The contents should be the same and there won't be
+        # any new request to the server
+        cache_issues = [cache_issue for cache_issue in jira.fetch_from_cache()]
+        del cache_issues[0]['timestamp']
+
+        expected_req = {
+                            'expand': ['renderedFields,transitions,operations,changelog'],
+                            'jql': [' updated > 0'],
+                            'startAt': ['0']
+                        }
+
+        request = httpretty.last_request()
+        self.assertEqual(request.method, 'GET')
+        self.assertRegex(request.path, '/rest/api/2/search')
+        self.assertDictEqual(request.querystring, expected_req)
+
+        self.assertEqual(len(issues), 1)
+        self.assertEqual(len(cache_issues), 1)
+
+        self.assertEqual(cache_issues, issues)
+
+        self.assertDictEqual(issues[0], json.loads(expected_json))
+
+    def test_fetch_from_cache_empty(self):
+        """Test if there are not any issues returned when the cache is empty"""
+
+        cache = Cache(self.tmp_path)
+
+        jira = Jira(JIRA_SERVER_URL, cache=cache)
+        cache_issues = [cache_issue for cache_issue in jira.fetch_from_cache()]
+
+        self.assertEqual(len(cache_issues), 0)
+
+    def test_fetch_from_non_set_cache(self):
+        """Test if a error is raised when the cache was not set"""
+
+        jira = Jira(JIRA_SERVER_URL)
+
+        with self.assertRaises(CacheError):
+            _ = cache_issues = [cache_issue for cache_issue in jira.fetch_from_cache()]
+
+class TestJiraBackendParsers(unittest.TestCase):
+    """ Jira backend parsers tests"""
+
+    def test_parse_questions(self):
+        """ Test issue parsing """
+
+        raw_parse_json = read_file('data/jira/jira_issues_page_1.json')
+        parse_json = read_file('data/jira/jira_issues_parse_expected.json')
+
+        issues = Jira.parse_issues(raw_parse_json)
+
+        result = [issue for issue in issues]
+
+        parse = json.loads(parse_json)
+        self.assertDictEqual(result[0], parse[0])
```

Comment:
I don't like this way of testing. For this tests, I prefer hardcoded values than compare results with files. I want to see what it is going on and what we are trying to test. Please remove this from the other tests and add some better checks.